File: src/main/java/arcana/common/items/aspect/Crystal.java
package arcana.common.items.aspect;

import arcana.common.aspects.Aspect;
import net.minecraft.item.Item;
import net.minecraft.item.ItemStack;
import net.minecraft.util.ResourceLocation;
import net.minecraft.util.text.ITextComponent;
import net.minecraft.util.text.TranslationTextComponent;

public class Crystal extends Item {
    public final Aspect aspect;
    public Crystal(Properties props, Aspect aspect){
        super(props);
        this.aspect = aspect;
        setRegistryName(new ResourceLocation(aspect.id.getNamespace(), aspect.id.getPath() + "_crystal"));
    }

    @Override
    public ITextComponent getName(ItemStack pStack) {
        return new TranslationTextComponent("item.arcana.crystal", new TranslationTextComponent(aspect.translationKey));
    }
}

File: src/main/kotlin/arcana/common/items/aspect/Crystal.kt
package arcana.common.items.aspect

import arcana.common.aspects.Aspect
import arcana.utils.Util.withPath
import net.minecraft.item.Item
import net.minecraft.item.ItemStack
import net.minecraft.util.text.TranslationTextComponent

class Crystal(props: Properties, val aspect: Aspect) : Item(props) {
    init {
        registryName = aspect.id.withPath{it+"_crystal"}
    }

    override fun getName(pStack: ItemStack) =
        TranslationTextComponent("item.arcana.crystal", TranslationTextComponent(aspect.translationKey))

}